src/mem.rs

```diff
@@ -0,0 +1,60 @@
+// NOTE Copied verbatim from the rlibc crate
```

Comment:
May not be the best idea. There's a bug for all of these when `n` > `ISIZE_MAX` that I've been trying to figure out how to fix to submit a PR.